verifyFiles.py

```diff
@@ -157,13 +189,29 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_fro
 		    elif conditions_dict[c][subject][file_name].st_size != conditions_dict[d][subject][file_name].st_size :
 		      diff[key][file_name]+=1
                       files_are_different=True
-	   	    elif checksum(abs_path_c) != checksum(abs_path_d):# TODO:when there are multiple conditions, we will compute checksums multiple times.We should avoid that.
-		      #Condition below makes sure that the checksums in the file after processing and in local are equal
-		      diff[key][file_name]+=1
-                      files_are_different=True
+		    elif abs_path_c in dictionary_checksum and  abs_path_d in dictionary_checksum:
+		      if is_checksum_equal(dictionary_checksum,abs_path_c,abs_path_d):
+		        diff[key][file_name]+=1
+			files_are_different=True
+		    elif abs_path_c not in dictionary_checksum and abs_path_d in dictionary_checksum:
+		      dictionary_checksum[abs_path_c]=checksum(abs_path_c)
+		      if is_checksum_equal(dictionary_checksum,abs_path_c,abs_path_d):
+			diff[key][file_name]+=1
+			files_are_different=True
+		    elif abs_path_c in dictionary_checksum and abs_path_d not in dictionary_checksum:
+		      dictionary_checksum[abs_path_d] = checksum(abs_path_d)
+		      if is_checksum_equal(dictionary_checksum,abs_path_c,abs_path_d):
+			diff[key][file_name]+=1
+			files_are_different=True
+		    else:
+		      dictionary_checksum[abs_path_d] = checksum(abs_path_d)
+		      dictionary_checksum[abs_path_c] = checksum(abs_path_c)
+                      if is_checksum_equal(dictionary_checksum,abs_path_c,abs_path_d):
+                        diff[key][file_name]+=1
+                        files_are_different=True
```

Comment:
It works (I guess) but this is not very elegant and there is a lot of redundant code there. A better solution is (pseudo code):
```
for filename in {abs_path_d,abs_pathc}:
   if dictionary_checksum[filename] is empty:
      dictionary_checksum[filename] = checksum(filename)
if is_checksum_equal(dictionary_checksum,abs_path_c,abs_path_d):
      diff[key][file_name]+=1
      files_are_different=True
```
That's about 6 lines where you need 19. Or did I miss anything?